Training/Training/Training.c

```diff
@@ -1,14 +1,123 @@
-// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
-// Copyright (c) Metamation India.
-// ------------------------------------------------------------------
-// Program.c
-// Program on main branch.
-// ------------------------------------------------------------------------------------------------
-
-#include<stdio.h>
-
-int main () {
-	printf ("Hello World!");
-	return 0;
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// Training.c
+// Executes the main function
+// Sudarson S
+// ------------------------------------------------------------------------------------------------
+#include"header.h"
+#include<stdbool.h>
+#include<stdio.h>
+#include<string.h>
+#include<errno.h>
+#include<malloc.h>
+#include<stdlib.h>
+
+/// <summary>Checks if a given input is a valid integer</summary>
+void IsValidInteger (char* prompt, int* num) {
+   bool output;
+   do {
+      char numberStr[13], * result = NULL;
+      printf ("%s", prompt);
+      fgets (numberStr, sizeof (numberStr), stdin);
+      result = strchr (numberStr, '\n');
+      if (result == NULL) {
+         while (getchar () != '\n');
+         output = false;
+      } else {
+         errno = 0;
+         char* endptr = NULL;
+         *num = strtol (numberStr, &endptr, 10);
+         output = endptr == numberStr || *endptr != '\n' || errno == ERANGE || numberStr[0] == ' ' ||
+            numberStr[0] == '\t' ? false : true;
+      }
+      if (output == false)
+         printf ("Invalid Input\n");
+   } while (output == false);
+}
+
+/// <summary>Accepts user inputs</summary>
+void UserInputs (int** arr, int* arrSize) {
+   do {
+      IsValidInteger ("\nEnter an array size: ", arrSize);
+      if (*arrSize <= 0)
+         printf ("Invalid Input\n");
+   } while (*arrSize <= 0);
+   do
+      *arr = (int*)malloc ((*arrSize) * sizeof (int));
+   while (*arr == NULL);
+   for (int i = 0; i < *arrSize; i++) {
+      int element;
+      IsValidInteger ("\nEnter an integer : ", &element);
+      (*arr)[i] = element;
+   }
+}
+
+/// <summary>Prints the array elements</summary>
+void PrintArray (int* arr, int n) {
+   for (int i = 0; i < n; i++)
+      printf ("%d ", arr[i]);
+}
+
+/// <summary>Runs test cases</summary>
+void TestCases () {
+   int inputs[][8] = { {0,12,3,4,4,4,4,4},{-98,45,2,1,3,4,0,0},{6,7,8,9,-4,-5,-9,0},{0,0,0,0,0,0,0,0} }, j,
+      binSearchKey[] = { 12,4,-4,6 }, binExpected[] = { 7,6,2,-1 }, n = sizeof (inputs) / sizeof (inputs[0]),
+      sortExpected[][8] = { { 0,3,4,4,4,4,4,12},{-98,0,0,1,2,3,4,45 },{-9,-5,-4,0,6,7,8,9 },{0,0,0,0,0,0,0,0} };
+   printf ("\nTestCases\n~~~~~~~~~");
+   for (int i = 0; i < n; i++) {
+      int output, flag = 0;

```

Comment:
Before commenting, please check if you have removed the unnecessary variables after making a change.